### PR TITLE
Use Yarn

### PR DESCRIPTION
**Pull request by jameskerr**
Replace npm with yarn version 3.0.1.

We now get to benefit from workspaces, constraints, a fast cache, better script output/ergonomics, and good docs. This PR should go in first, then I'll add the zealot repo as a workspace on top of this.

### CODE REVIEW

File: .github/workflows/advance-zed.yml [hunk 1]
```diff
@@ -79,22 +79,22 @@ jobs:
...
-      - run: npm install https://github.com/brimdata/zed#${{ steps.zed_pr.outputs.sha }}
-      - run: npm install --no-audit
-      - run: npm run build
-      - run: npm run lint
-      - run: npm run tsc
-      - run: npm test -- --maxWorkers=2 --ci
-      - run: npm run test:api
-      - run: npm run test:system
+      - run: yarn install https://github.com/brimdata/zed#${{ steps.zed_pr.outputs.sha }}
+      - run: yarn
```
Comment (nwt) [inline .github/workflows/advance-zed.yml:85]: Adding `--inline-builds` here prints build output for dependencies, which is really helpful when one fails.
```suggestion
      - run: yarn --inline-builds
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -21,16 +21,16 @@ jobs:
...
           node-version-file: .nvmrc
-      - run: npm install --no-audit
-      - run: npm run build
-      - run: npm run lint
-      - run: npm run tsc
-      - run: npm test --maxWorkers=2 --ci
-      - run: npm run test:api
-      - run: npm run test:system
-      - run: npm run generate-responses -- --silent
+      - run: yarn
```
Comment (nwt) [inline .github/workflows/ci.yml:26]:
```suggestion
      - run: yarn --inline-builds
```

File: .github/workflows/linux-release-candidate.yml [hunk 3]
```diff
@@ -5,65 +5,65 @@ on:
...
+        with:
+          go-version: "1.16"
+      - uses: actions/setup-node@v2
+        with:
+          cache: yarn
+          node-version-file: .nvmrc
+      - if: ${{ !startsWith(steps.extract_branch.outputs.branch, 'rc-v') }}
+        name: Set version string for ad hoc builds
+        run: echo "ADHOC_VERSION_STRING=$(npm version preminor --no-git-tag-version --preid prerelease-${GITHUB_SHA:0:8})" >> $GITHUB_ENV
+      - run: yarn
```
Comment (nwt) [inline .github/workflows/linux-release-candidate.yml:31]:
```suggestion
      - run: yarn --inline-builds
```

File: .github/workflows/macos-release-candidate.yml [hunk 4]
```diff
@@ -5,101 +5,101 @@ on:
...
+        with:
+          go-version: "1.16"
+      - uses: actions/setup-node@v2
+        with:
+          cache: yarn
+          node-version-file: .nvmrc
+      - if: ${{ !startsWith(steps.extract_branch.outputs.branch, 'rc-v') }}
+        name: Set version string for ad hoc builds
+        run: echo "ADHOC_VERSION_STRING=$(npm version preminor --no-git-tag-version --preid prerelease-${GITHUB_SHA:0:8})" >> $GITHUB_ENV
+      - run: yarn
```
Comment (nwt) [inline .github/workflows/macos-release-candidate.yml:29]:
```suggestion
      - run: yarn --inline-builds
```

File: .github/workflows/win-release-candidate.yml [hunk 5]
```diff
@@ -5,83 +5,83 @@ on:
...
+          go-version: "1.16"
+      - uses: actions/setup-node@v2
+        with:
+          cache: yarn
+          node-version-file: .nvmrc
+      - if: ${{ !startsWith(steps.extract_branch.outputs.branch, 'rc-v') }}
+        name: Set version string for ad hoc builds
+        run: echo "ADHOC_VERSION_STRING=$(npm version preminor --no-git-tag-version --preid prerelease-${GITHUB_SHA:0:8})" >> $GITHUB_ENV
+        shell: bash
+      - run: yarn
```
Comment (nwt) [inline .github/workflows/win-release-candidate.yml:31]:
```suggestion
      - run: yarn --inline-builds
```

**Review by mason-fish — APPROVED** [review]
Right on! 🧶

**Pull request merged**